pkg/analyzer/pod.go

```diff
@@ -83,14 +73,17 @@ func (PodAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) {
 							Sensitive: []common.Sensitive{},
 						})
 					}
-				}
-
-				// This represents container that is in CrashLoopBackOff state due to conditions such as OOMKilled
-				if containerStatus.State.Waiting.Reason == "CrashLoopBackOff" {
+				} else if containerStatus.State.Waiting.Reason == "CrashLoopBackOff" && containerStatus.LastTerminationState.Terminated != nil {
```

Comment:
What is the impact of this change? Would this still let us capture a pod that has restarted X times?

Reply:
I can't see anything wrong with it, but it's quite hard to read the code and be certain about all the code paths. I suggest we test this out a bit  - is there a test in place for every type of pod event? e.g. OOMKilled etc

Reply:
I see, and I too believe that we should spend some time to test it. Currently, not all pod events have been included in the tests. I'll try to include those events and push the updated tests soon